src/libertem/api.py

```diff
@@ -158,34 +183,89 @@ def make_with(cls, executor_spec: ExecutorSpecType, *args, **kwargs) -> 'Context
             "pipelined":
                 Create a :class:`~libertem.executor.pipelined.PipelinedExecutor`,
                 which is suitable for multi-process streaming live processing
-                using `LiberTEM-live <https://libertem.github.io/LiberTEM-live/>`_.
-        *args, **kwargs
-            Passed to :class:`Context`.
+                using `LiberTEM-live <https://libertem.github.io/LiberTEM-live/>`_
+        cpus : int | Iterable[int], optional
+            The number of CPU workers to create, where possible. The meaning of
+            CPU worker depends on the type of executor created - threaded executors
+            will interpret :code:`cpus` to choose the number of threads, while process-based
+            executors will interpret the value to choose the number of processes to spawn.
+            The iterable form of of the argument is intended contain CPU-id values to
+            enable cpu-pinning where the exector and the system support it, but most
+            use cases will only require an integer argument. Executors where this
+            parameter does not make sense will raise an error if provided.
+        gpus : int | Iterable[int], optional
+            Similar to :code:`cpus`, specifies the number of GPU workers to create
+            where the executor chosen supports it, else raise an error. The integer
+            form of the argument assigns workers round-robin to available GPUs on
+            the system, while the iterable form allows specifiying the number of
+            workers per GPU by repeating the id of any given GPU in the argument.
```

Comment:
The docstring is coherent with the current behaviour of `spec_cluster` (an int argument with no other params is simple round-robin assignment). Do we want to document the planned future behaviour at this time?

Reply:
Currently I believe that the workers are best distributed so that they have even and sufficient GPU RAM. The current behavior is round-robin. In order to keep this function backwards-compatible, we shouldn't specify a behavior until we are sure that whatever we choose will stick.

Reply:
Should be resolved now, I've made the gpu docstring non-commital.